ledger/applications.go

```diff
@@ -89,12 +100,17 @@ func newAppLedger(balances transactions.Balances, params basics.AppParams, white
 	al = &appLedger{}
 	al.appIdx = appIdx
 	al.balances = balances
-	al.addresses = make(map[basics.Address]bool)
-	al.params = params
-	for _, addr := range whitelist {
+	al.addresses = make(map[basics.Address]bool, len(acctWhitelist))
```

Comment:
in theory this can be done by `InitLedger` instead of `appGlobalWhitelist := append(ac.ForeignApps, appIdx)` so that we'll economize on these two iterations.